mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala

```diff
@@ -671,7 +671,7 @@ class ALS(@Since("1.4.0") override val uid: String) extends Estimator[ALSModel]
     instr.logDataset(dataset)
     instr.logParams(this, rank, numUserBlocks, numItemBlocks, implicitPrefs, alpha, userCol,
       itemCol, ratingCol, predictionCol, maxIter, regParam, nonnegative, checkpointInterval,
-      seed, intermediateStorageLevel, finalStorageLevel)
+      seed, intermediateStorageLevel, finalStorageLevel, coldStartStrategy)
```

Comment:
`coldStartStrategy` doesn't affect fitting the model, so I'm not sure about logging it here. I agree with `validationTol` though. I actually wouldn't have added `validationIndicatorCol` but it's in GBTClassifier, so yes stay consistent.

Reply:
Yes, we could let coldStartStrategy alone. BTW, I made a rapid scan and found that some algs do not log the columns.